numpy/core/numeric.py

```diff
@@ -650,23 +653,26 @@ def require(a, dtype=None, requirements=None):
     if not requirements:
         return asanyarray(a, dtype=dtype)
 
+    requirements = set(requirements)
```

Comment:
True, It isn't as strict, but simpler; I could go either way. The set is mostly useful to remove repeats and checking subsets, if you retain the long form that becomes more work.